CDS/scripts/js_utils/youtube/youtube-upload-lib.js

```diff
@@ -65,37 +65,58 @@ function getYoutubeData(connection) {
     });
 }
 
+function addPosterImage(connection, videoId, youtubeClient, account) {
+    return dataStore.get(connection, 'meta', 'poster_image')
+    .then(file => {
+
+        if (file.value) {
+
+          if (!account) {
+            return new Promise((fulfill, reject) => { reject( new Error('could not add a poster image: missing account owner') )});
+          }
+          return  new Promise((fulfill, reject) => {
+            youtubeClient.thumbnails.set({ videoId: videoId, onBehalfOfContentOwner: account, media: {body: fs.createReadStream(file.value)}}, (err, result) => {
+                  if (err) reject(err);
+                  else fulfill(result);
+            });
+          });
+        }
+        return new Promise(fulfill => {fulfill()});
```

Comment:
For clarity, I could rename the function to indicate that the addition of the poster image is conditional, then it would make sense for the promise to resolve even if the poster image did not exist?